### PR TITLE
Add Sort.java to the java API

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR adds Sort.java SortTest.java and cvc5_Sort.cpp to the java api.

### CODE REVIEW

**Comment by mudathirmahgoub** [comment]
This PR depends on (#6196).
To make the PR small, the files are not included yet in the build. They are included in this [branch](https://github.com/mudathirmahgoub/CVC4/tree/sort)

File: test/unit/api/java/cvc5/SortTest.java [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+  }
+
+  @Test void operators_comparison()
+  {
+    assertDoesNotThrow(() -> d_solver.getIntegerSort() == d_solver.getNullSort());
+    assertDoesNotThrow(() -> d_solver.getIntegerSort() != d_solver.getNullSort());
+    assertDoesNotThrow(() -> d_solver.getIntegerSort().compareTo(d_solver.getNullSort()));
+    //   assertDoesNotThrow(() -> d_solver.getIntegerSort() <=d_solver.getNullSort());
+    //   assertDoesNotThrow(() -> d_solver.getIntegerSort() >d_solver.getNullSort());
+    //   assertDoesNotThrow(() -> d_solver.getIntegerSort() >=d_solver.getNullSort());
```
Comment (4tXJ7f) [inline test/unit/api/java/cvc5/SortTest.java:70]: Let's remove those instead of commenting them out

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
LGTM, just a minor request for changes.

File: test/unit/api/java/cvc5/SortTest.java (hunk 1 above)
Comment (mudathirmahgoub) [reply test/unit/api/java/cvc5/SortTest.java]: Done

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the updates.

**Pull request merged**